### PR TITLE
[Upsert] Do not require ON CONFLICT clause on INSERT OR REPLACE in some situations.

### LINKED ISSUE #6576: Conflict target required when table has multiple primary key columns
### What happens?

`insert or replace` on a table with multiple primary keys requires a conflict_target. However the [docs](https://duckdb.org/docs/sql/statements/insert#on-conflict-clause) state it's optional and *all* keys will be targeted when there's no conflict_target:

> Optionally you can provide a conflict_target, which is a group of columns that an Index indexes on, or if left out, all UNIQUE or PRIMARY KEY constraint(s) on the table are targeted.

### To Reproduce

Without a conflict target,
```
D create table foo(a int, b int, c int, d int, primary key(a,b,c));
D insert or replace into foo values (1,2,3,4);
Error: Binder Error: Conflict target has to be provided for a DO UPDATE operation when the table has multiple UNIQUE/PRIMARY KEY constraints
```

Only works with a conflict target, eg:
```
D insert into foo values (1,2,3,4) on conflict(a,b,c) do update set d=4;
```

### OS:

macos

### DuckDB Version:

v0.7.1 b00b93f0b1

### DuckDB Client:

cli

### Full Name:

Oliver Mannion

### Affiliation:

Xero

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #6576 

Previously we threw an error even when only a single index exists on the table if it contained multiple columns.

### CODE REVIEW

File: test/sql/upsert/insert_or_replace_ambiguity.test [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+statement ok
+create table foo(
+	a int,
+	b int,
+	c int,
+	primary key(a,b,c)
+);
+
+statement ok
+insert or replace into foo values (1,2,3);
```
Comment (taniabogatsch) [inline test/sql/upsert/insert_or_replace_ambiguity.test:14]: Can we perform the inserts twice and also check the table data after? And let's do the same for the other tests.

File: test/sql/upsert/insert_or_replace_ambiguity.test [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# Single UNIQUE CONSTRAINT
+statement ok
+create table foo4(
+	a int unique,
+	b int
+);
+
+statement ok
+insert or replace into foo4 values(1, 2);
```
Comment (taniabogatsch) [inline test/sql/upsert/insert_or_replace_ambiguity.test:57]: Isn't this the same or am I missing something here?

**Review by taniabogatsch — CHANGES_REQUESTED** [review]
Looks good! I only had to minor comments on the tests.

File: test/sql/upsert/insert_or_replace_ambiguity.test (hunk 2 above)
Comment (Tishj) [reply test/sql/upsert/insert_or_replace_ambiguity.test:57]: Yes they are, I think I copied something incorrectly?

Supposed to have a UNIQUE INDEX created with a separate statement after the create
The `Single UNIQUE CONSTRAINT` test is correct 👍

**Review by taniabogatsch — APPROVED** [review]

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**